plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/swarm/client/AwsEcrAuthResolver.java

```diff
@@ -0,0 +1,128 @@
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.ecr.AmazonECRClient;
+import com.amazonaws.services.ecr.model.AmazonECRException;
+import com.amazonaws.services.ecr.model.AuthorizationData;
+import com.amazonaws.services.ecr.model.GetAuthorizationTokenRequest;
+import com.amazonaws.services.ecr.model.GetAuthorizationTokenResult;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.plugin.docker.client.DockerRegistryDynamicAuthResolver;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+/**
+ * This class provides actual auth config for AWS ECR.
+ *
+ * @author Mykola Morhun
+ */
+@Singleton
+public class AwsEcrAuthResolver implements DockerRegistryDynamicAuthResolver {
+    private static final Logger LOG = LoggerFactory.getLogger(DockerRegistryDynamicAuthResolver.class);
+
+    private final AwsInitialAuthConfig awsInitialAuthConfig;
+
+    @Inject
+    public AwsEcrAuthResolver(AwsInitialAuthConfig awsInitialAuthConfig) {
+        this.awsInitialAuthConfig = awsInitialAuthConfig;
+    }
+
+    /**
+     * Retrieves actual auth data for Amazon ECR.
+     * Returns null if no credential configured for specified registry.
+     * Note, that credentials is changed every 12 hours.
+     *
+     * @return actual auth config for specified AWS ECR or null if no credentials configured
+     */
+    @Override
+    public AuthConfig getDynamicXRegistryAuth(String registry) {
+        if (registry == null || !registry.equals(awsInitialAuthConfig.getEcr())) {
+            return null;
+        }
+
+        String authorizationToken = getAwsAuthorizationToken();
+        if (authorizationToken == null) {
+            return null;
+        }
+        String decodedAuthorizationToken;
+        try {
+            decodedAuthorizationToken = new String(Base64.getDecoder().decode(authorizationToken));
+        } catch (IllegalArgumentException e) {
+            LOG.warn("Retrieved AWS ECR authorization token has invalid format");
+            return null;
+        }
+        int colonIndex = decodedAuthorizationToken.indexOf(':');
+        if (colonIndex == -1) {
+            LOG.warn("Cannot retrieve ECR credentials from token");
+            return null;
+        }
+
+        return newDto(AuthConfig.class).withUsername(decodedAuthorizationToken.substring(0, colonIndex))
+                                       .withPassword(decodedAuthorizationToken.substring(colonIndex + 1));
+    }
+
+    /**
+     * Retrieves actual auth config for Amazon ECR.
+     * Returns null if no AWS ECR credentials configured.
+     *
+     * @return actual AWS ECR auth config or empty map if ECR not configured
```

Comment:
Please avoid duplicate description of method. Differentiate provided explanation or remove one of the descriptions